sc-locked-asset-factory/src/lib.rs

```diff
@@ -118,11 +131,16 @@ pub trait LockedAssetFactory:
 
         let locked_remaining = amount.clone() - unlock_amount;
         if locked_remaining > 0 {
-            let new_unlock_milestones = self
-                .create_new_unlock_milestones(current_block_epoch, &attributes.unlock_milestones);
+            let new_unlock_milestones = self.create_new_unlock_milestones(
+                current_block_epoch,
+                &unlock_schedule.unlock_milestones,
+            );
+            let new_unlock_schedule = UnlockSchedule {
+                unlock_milestones: new_unlock_milestones,
+            };
             let _ = self.produce_tokens_and_send(
                 &locked_remaining,
-                &new_unlock_milestones,
+                &new_unlock_schedule,
```

Comment:
How is this computed : I have 100 Locked MEX, I unlock after first period end and will receive 10MEX and 90 locked mex with release schedule of 15 percent for the next 6 months. After one month I unlock_assets again. 15% of 90 = 13.5 MEX and I should have received 15. In the end if I unlock at every month I will get much less tokens.

Reply:
The rest of percents are recalculated when unlocking. A user won't get 15% from 90. He will get the equivalent of 15% from the initial value.